src/templates/csharp/TestBase.cs

```diff
@@ -105,133 +105,149 @@ public void SwitchToNativeContext()
-                        if (textAttr == null)
-                            textAttr = "";
-                        textAttr = textAttr.Trim().ToLower();
-                        if (!string.IsNullOrEmpty(textAttr))
-                            nativeTexts.Add(textAttr);
-                    }
+                    if (element.NodeType != XmlNodeType.Element) continue;
+                    string? textAttr = element.Attributes?[isIos ? "value" : "text"]?.Value;
+                    if (textAttr == null)
+                        textAttr = "";
+                    textAttr = textAttr.Trim().ToLower();
+                    if (!string.IsNullOrEmpty(textAttr))
+                        nativeTexts.Add(textAttr);
                 }
+            }
 
-                HashSet<string> contexts = new HashSet<string>(driver.Contexts);
-                foreach (string context in contexts)
+            var contexts = driver.Contexts;
+            foreach (var context in contexts)
+            {
+                if (context.StartsWith("WEBVIEW") || context.Equals("CHROMIUM"))
                 {
-                    if (context.StartsWith("WEBVIEW") || context.Equals("CHROMIUM"))
+                    string source = null;
+                    try
                     {
-                        string source = null;
-                        try
-                        {
-                            SwitchContext(context);
-                            source = driver.PageSource;
-                        }
-                        catch (Exception ex)
-                        {
-                            Console.WriteLine($"Bad context {context}, error \"{ex.Message}\", skipping...");
-                            continue;
-                        }
-
-                        if (source == null) continue;
-                        ContextInfo contextInfo = contextInfos.FirstOrDefault(e => e.context.Equals(context));
-                        if (contextInfo == null)
-                        {
-                            contextInfo = new ContextInfo(context);
-                            contextInfos.Add(contextInfo);
-                        }
-
-                        contextInfo.sourceLength = source.Length;
-                        if (nativeTexts.IsNullOrEmpty()) continue;
+                        SwitchContext(context);
+                        source = driver.PageSource;
+                    }
+                    catch (Exception ex)
+                    {
+                        Log($"Bad context {context}, error \"{ex.Message}\", skipping...");
+                        continue;
+                    }
 
-                        HtmlDocument htmlDoc = LoadXMLFromString(source);
-                        HtmlNode bodyElements = htmlDoc.DocumentNode.SelectSingleNode("/html/body");
-                        if (bodyElements == null) continue;
+                    if (source == null) continue;
```

Comment:
For your reference, we can add one more valuable condition here to get rid of invalid/hidden web contexts, just execute the script `document.hidden` to check if it's visible or not.
Ref:
- https://github.com/kobiton/ita-core-util/blob/main/src/screen-info-builder/index.ts#L448
- https://github.com/kobiton/static-resources-manager/blob/master/src/bin/ksl-files/kobiton-inventory.js#L62

Reply:
I will apply this logic on next PR. Thank you!